src/write_riemann.c

```diff
@@ -616,16 +616,16 @@ static void wrr_free(void *p) /* {{{ */
     return;
 
   pthread_mutex_lock(&host->lock);
-
   host->reference_count--;
-  if (host->reference_count > 0) {
-    pthread_mutex_unlock(&host->lock);
+  int reference_count = host->reference_count;
+  pthread_mutex_unlock(&host->lock);
+
+  if (reference_count > 0) {
```

Comment:
What's the reason for this change, doesn't it introduce small chance for race condition?

PS.  See also https://github.com/collectd/collectd/issues/4050

Reply:
I wanted to keep this code consistent across plugins. This looks like an oversight. Fixed.